sps30-i2c/sps30.c

```diff
@@ -70,27 +72,19 @@ int16_t sps30_read_firmware_version(uint8_t *major, uint8_t *minor) {
 }
 
 int16_t sps30_get_serial(char *serial) {
-    uint16_t i;
-    int16_t ret;
-    union {
-        char serial[SPS30_MAX_SERIAL_LEN];
-        uint16_t __enforce_alignment;
-    } buffer;
-
-    ret = sensirion_i2c_read_cmd(SPS30_I2C_ADDRESS, SPS_CMD_GET_SERIAL,
-                                 (uint16_t *)buffer.serial,
-                                 SENSIRION_NUM_WORDS(buffer.serial));
-    if (ret != STATUS_OK)
-        return ret;
+    int16_t error;
 
-    SENSIRION_WORDS_TO_BYTES(buffer.serial, SENSIRION_NUM_WORDS(buffer.serial));
-    for (i = 0; i < SPS30_MAX_SERIAL_LEN; ++i) {
-        serial[i] = buffer.serial[i];
-        if (serial[i] == '\0')
-            return 0;
-    }
+    error = sensirion_i2c_write_cmd(SPS30_I2C_ADDRESS, SPS_CMD_GET_SERIAL);
 
-    return 0;
+    if (error != NO_ERROR)
+        return error;
+
+    error = sensirion_i2c_read_words_as_bytes(
+        SPS30_I2C_ADDRESS, (uint8_t *)serial, SPS30_SERIAL_NUM_WORDS);
+
+    // TODO Final '\0'?
```

Comment:
In the SCD driver we add a final `'\0'` in every case. But this would mean that the buffer needs to be allocated for 33 bytes, not 32 bytes. But I'm not sure if it is guaranteed from the SHDLC protocal that strings get sent with a terminating `'\0'`.
Maybe @fuerstr or @ubruhin can comment on this?

Reply:
Resolved offline: The FW should always set any unused bytes of the serial number to `'\0'`. Also the serial number should never be longer than 32 bytes including the terminating `'\0'`.